resources/function_help/json/if

```diff
@@ -4,10 +4,11 @@
   "description": "Tests a condition and returns a different result depending on the conditional check.",
   "arguments": [
 	{"arg":"condition","description":"the condition which should be checked"},
-	{"arg":"result_when_true","description":"the result which will be returned when the condition is True"},
-	{"arg":"result_when_false","description":"the result which will be returned when the condition is False"}
+	{"arg":"result_when_true","description":"the result which will be returned when the condition returns True or not null value"},
```

Comment:
I also a better more simple example is in order:

1 + 1 = 2 = True
1 + 1 = 3 = False

I think that is a little easier for people to reason about then a single
if(trueishvalue)

On Wed, Mar 30, 2016 at 6:03 PM, Matthias Kuhn <notifications@github.com>
wrote:

> In resources/function_help/json/if
> <https://github.com/qgis/QGIS/pull/2953#discussion_r57849766>:
>
> > @@ -4,10 +4,11 @@
> >    "description": "Tests a condition and returns a different result depending on the conditional check.",
> >    "arguments": [
> >  	{"arg":"condition","description":"the condition which should be checked"},
> > -	{"arg":"result_when_true","description":"the result which will be returned when the condition is True"},
> > -	{"arg":"result_when_false","description":"the result which will be returned when the condition is False"}
> > +	{"arg":"result_when_true","description":"the result which will be returned when the condition returns True or not null value"},
>
> A condition does not "return" (a function does), a condition "is" or
> "evaluates to". To improve the docs here it would IMO be better to be more
> precise on what True means (like done in the second example). In general
> it's also not "not null value" but "not null".
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/qgis/QGIS/pull/2953/files/01e0395a2fc218ba35e1e89a2e407d4d5f554ae8#r57849766>
>